packages/protocol/contracts/layer1/provers/ProverSet.sol

```diff
@@ -94,18 +94,22 @@ contract ProverSet is EssentialContract, IERC1271 {
         LibAddress.sendEtherAndVerify(admin, _amount);
     }
 
-    /// @notice Proposes a block only when it is the first block proposal in the current L1 block.
-    function proposeBlockV2Conditionally(
-        bytes calldata _params,
-        bytes calldata _txList
+    /// @notice Proposes a batch blocks only when it is the first batch blocks proposal in the
+    /// current L1 block.
+    function proposeBlocksV2Conditionally(
+        bytes[] calldata _params,
+        bytes[] calldata _txList
     )
         external
         onlyProver
     {
         ITaikoL1 taiko = ITaikoL1(taikoL1());
-        // Ensure this block is the first block proposed in the current L1 block.
-        require(taiko.lastProposedIn() != block.number, NOT_FIRST_PROPOSAL());
-        taiko.proposeBlockV2(_params, _txList);
+
+        uint256 lastProposedBlock = taiko.lastProposedIn();
```

Comment:
I think the logic here is the same right? If so, I think we should revert changes in this function.

Reply:
Yes, this change is made by @swarna1101 , maybe he can give some reasons.

> I think the logic here is the same right? If so, I think we should revert changes in this function.